### DISCUSSION TITLE
[Bug?] private members are being listed in auto-completion.

### DISCUSSION THREAD

**Discussion by Unknown6656** [dotnet/roslyn#71432] [General] [answered]
I noticed a potential bug occurring when developing code generators: The intellisense auto-complete would enumerate private (and compiler-generated) members in the list of possible completions, .e.g. the private nested class `<>c`:
![image](https://github.com/dotnet/csharplang/assets/8807985/6e60b4e8-e9e3-4e49-85b7-0b17fb58db15)

I tried to replicate that with [a minimal working sample](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4HtgCsCmBjAFwAIAPIgXiJgFcAbWgbgFgAoYddWo4C0oqAM5EAwgDoiAegkAeALTyiAB1oQ8OIgQAW6vNQBOA9HqIQAZgRzGt6hOmIQYCJXpwChAbQDKACQCSAMQAVAF1kd2FAgCUAGWDWIgSiVlYAAQBmIhSAJhEiAG94xMU9KAA3CAtMnIAhZhYAXyA===) and got the same behaviour:
![image](https://github.com/dotnet/csharplang/assets/8807985/644c6b21-0c64-4ce7-a2ca-9a88efb6021b)

Now my question is simply the following:

### Is that intentional or is that a bug?
It kinda feels "illegal", as the private nested class is not accessible:

> error CS0122: 'C.B' is inaccessible due to its protection level

**Comment by CyrusNajmabadi** [comment]
What tool are you using. That doesn't really look like VS.

**Reply by Unknown6656** [reply]
I'm using VS2022 (17.9.0 Preview 2.0) in the screenshot above. I had to blur out the name of the method due to confidentiality reasons. The second screenshot was taken in the current version of sharplab.io

**Reply by CyrusNajmabadi** [reply]
can you supply a repro?  I can't repro this myself.  Sharplab is different here.  Thanks!

**Reply by sharwell** [reply]
Here's a better repro:
https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4HtgCsCmBjAFwAIAPIgXiJgFcAbWgbgFgAoYddWo4C0oqAM5EAwgyIB6cQB4AtHKIAHWhDw4iBABZq81AE4D0uohABmBHEeHGYCRbpwChAOlasAAgGYibgEwiiAN6sRCF2UABuEObefgBCgUQAvqyJQA

**Reply by CyrusNajmabadi** [reply]
Sam, do you have a repro where `<>c` shows up?

**Comment by CyrusNajmabadi** [comment, marked as answer]
Fixed with https://github.com/dotnet/roslyn/pull/71489.

**Reply by Unknown6656** [reply]
thank you very much for the speedy bug fix :)
